coderd/workspaces_test.go

```diff
@@ -340,6 +340,99 @@ func TestWorkspace(t *testing.T) {
 	})
 }
 
+func TestResolveAutostart(t *testing.T) {
+	t.Parallel()
+
+	t.Run("OK", func(t *testing.T) {
+		t.Parallel()
+		ownerClient := coderdtest.New(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
```

Comment:
Can you do this without a ProvisionerDaemon following the new `dbfake` paradigm introduced here: https://github.com/coder/coder/pull/10426 ?

Reply:
I can make this change but it's not as trivial as it sounds especially because of parameters. Once I finish up this feature I'm planning on helping refactor existing tests using `dbfake`. Do you care if we push it for now?

Reply:
I'm wary of shipping a feature with a "chore" task leftover that we pinky-promise to fix up soon.  Once the feature is shipped it's real easy to just put off the chore indefinitely.  Even if you created a GitHub ticket, we auto-close stale issues!  I'm not questioning your sincerity, but the road to hell is paved with best intentions.

I think debt is a good analogy in this case.  You're saying we should take out a loan here.  What's so important about shipping this change now such that we want to take the risk of not fixing the tests?  Flaky tests are like, the biggest thing on developer's mind judging by our conversation at the offsite.

Reply:
Yeah agreed, per our offline conversation I'm going to refactor these in a separate PR.